modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -170,21 +178,13 @@ private MetadataContext getMetadataContext() throws MetadataResolvingException
     {
         return getConfigurationProvider()
                 .map(provider -> provider.get(event))
-                .orElseGet(() -> {
-                    if (StringUtils.isBlank(configurationProviderName))
-                    {
-                        return extensionManager.getConfiguration(extensionModel, event);
-                    }
-                    return extensionManager.getConfiguration(configurationProviderName, event);
```

Comment:
`extensionManager.getConfiguration(configurationProviderName, event);` is redundant since that case should already be covered by `getConfigurationProvider()`